proposals/NNNN-Add-screen-stack-management-module.md

```diff
@@ -0,0 +1,96 @@
+
+
+## Motivation
+Currently, the screens displayed by `Show` and `PerformInteraction` are independent, and there is no relation between each screen. However, in many cases of the App development process, each screen has a relationship with each other. Therefore, the screen transition specifications expected by the app developers may not be realized. By adding a screen management module and managing transitions, this problem can be solved and reduce the burden to screen development. For example, pressing the back button to enable screen transitions such as return to the previous screen. As a result, it is possible to build a better UX by reducing the inconvenience of use as much as possible for the users.
+
+<b>Current behavior</b>
+ - `PerformInteraction` and `Show (template)` are independent screens.
+ - HMI controls menu/submenu displays and screen transitions.
+ - Contents displayed in menu/submenu are provided by `AddCommand`, `DeleteCommand`, `AddSubMenu`, `DeleteSubMenu` of the app.
+ - When pushing the menu item, the HMI sends an `onCommand` notification to the mobile, and then the mobile determines the operation.
+   After that, you cannot return to the menu screen, but you will return to the first template screen.
+ - Even if the `DeleteSubMenu` is notified from mobile while the submenu is displayed, it is rejected.
+
+<b>Current Screen transition process</b>
+
+- RPC related to menu/submenu are sent after the successful registration of app.
+```
+AddCommand(commandID = 1，"command1")
+AddSubMenu(menuID = 100, "submenu1")
+AddCommand(commandID = 2，"command2")
+AddCommand(commandID = 4，"command4", parentID = 100)
+AddCommand(commandID = 5，"command5", parentID = 100)
+AddCommand(commandID = 6，"command6", parentID = 100)
+```
+
+- RPC for `PerformInteraction`
+```
+CreateInteractionChoiceSet[
+	 interactionChoiceSetID = 1000,
+	(choiceID = 1,  menuName = "choice1"),
+	(choiceID = 2,  menuName = "choice2"),
+	(choiceID = 3,  menuName = "choice3") ]
+```
+
+![Screen_transition_process.png](../assets/proposals/NNNN-Add-screen-stack-management-module/Screen_transition_process.png)
+
+## Proposed solution
+To solve this problem, build the following screen management mechanism.
+
+- Add a screen management module, screenTaskManager, to SDL app library to manage screen transition stacks such as menu, submenu, `PerformInteraction` displays, popo-ups, etc.
+- Add `button Name` for subscribe.
+- Add screen change notification process `notifyScreenChanged` to SDL app library.
+- Add the screen management module to [the iOS Guides](https://smartdevicelink.com/en/guides/iOS/displaying-a-user-interface/template-images/), [the Android Guides](https://smartdevicelink.com/en/guides/android/displaying-a-user-interface/main-screen-templates/) and [the JavaScript Guides](https://smartdevicelink.com/en/guides/javascript/displaying-a-user-interface/main-screen-templates/).
```

Comment:
```suggestion
- Add the screen management module to [the iOS Guides](https://smartdevicelink.com/en/guides/iOS/displaying-a-user-interface/template-images/), [the Android Guides](https://smartdevicelink.com/en/guides/android/displaying-a-user-interface/main-screen-templates/) and [the JavaScript Suite Guides](https://smartdevicelink.com/en/guides/javascript/displaying-a-user-interface/main-screen-templates/).
```
Do the JavaSE and JavaEE guides need to be updated as well?  If so, please add here.  Thanks!

Reply:
Hi, @theresalech -san.
Thank you in advance for your reply.
Regarding to the JavaScript Suite, I will modify it and make a commit it.
On the other hand, regarding to JavaSE and JavaEE, since I think these are the targets, I will modify this sentence to "SDL app library's documentations" and make a commit it.
Could you please condirm it.
By the way, I do not know what devices are JavaSE and JavaEE implemented.
So, could you please teach me those target devices.
Thank you.

Reply:
I've provided a new suggestion for this section.  Regarding JavaSE and JavaEE, please find guides here: https://smartdevicelink.com/en/guides/javase/getting-started/installation/, https://smartdevicelink.com/en/guides/javaee/getting-started/installation/.

Reply:
Hi, @theresalech -san.
Thank you for your response.
I confirmed the Java SE Guides and the Java EE Guides.
But, I did not understand because there was no description the target devices.
For example, I know that the iOS Guides are for the apple mobile devices, the Android Guides are for the Android mobile devices and the JavaScript Suite Guides are for the cloud system.
What devices are the Java SE Guides for?
What devices are the Java EE Guides for?
Could you please give me an advice?
Thank you.

Reply:
@Akihiro-Miyazaki this question is best suited for the [#android-help](https://smartdevicelink.slack.com/archives/C0K2W1V28) channel on the SmartDeviceLink Slack team.  Thanks!

Reply:
@theresalech -san,
Thank you for your comment.